data-prepper-plugins/kafka-source/src/main/java/org/opensearch/dataprepper/plugins/consumer/MultithreadedConsumer.java

```diff
@@ -0,0 +1,204 @@
+					ConsumerRecords<String, String> records = consumer.poll(Duration.ofMillis(100));
+					if (records.count() > 0) {
+						for (TopicPartition partition : records.partitions()) {
+							List<ConsumerRecord<String, String>> partitionRecords = records.records(partition);
+							for (ConsumerRecord<String, String> consumerRecord : partitionRecords) {
+								currentOffsets.put(
+										new TopicPartition(consumerRecord.topic(), consumerRecord.partition()),
+										new OffsetAndMetadata(consumerRecord.offset() + 1, null));
+								publishRecordToBuffer(generateStringEventRecord(consumerRecord.value()), consumer,
+										partition, consumerRecord, lastReadOffset);
+							}
+							lastReadOffset = partitionRecords.get(partitionRecords.size() - 1).offset();
+							commitOffsets(partition, lastReadOffset, consumer);
+							recordCount.incrementAndGet();
+						}
+					}
+				}
+			} else {
+				logger.info("The given TOPIC : {} is empty and couldn't be able to read the records...",
+						sourceConfig.getTopicName());
+			}
+		} catch (Exception exp) {
+			if (exp.getCause() instanceof WakeupException && !status.get()) {
+				logger.error("Error reading records from the topic...{}", exp.getMessage());
+			}
+
+		} finally {
+			logger.info("Closing the consumer... {}", consumerId);
+			consumer.close();
+		}
+	}
+
+	
+	public void shutdownConsumer() {
+		status.set(false);
+		consumer.wakeup();
+	}
+	
+	
+	@SuppressWarnings("deprecation")
+	private Record<Event> generateStringEventRecord(String evenRrecord) {
+		final Event event = JacksonEvent.fromMessage(evenRrecord);
+		return new Record<>(event);
+	}
+
+	
+	@SuppressWarnings({ "deprecation" })
+	private void publishRecordToBuffer(Record<Event> eventRecord, KafkaConsumer<String, String> consumer,
+			TopicPartition partition, ConsumerRecord<String, String> consumerRecord, long lastReadOffset) {
+		try {
+			buffer.write(eventRecord, 500);
+		} catch (Exception exp) {
+			logger.error("Error while writing records to the buffer {}", exp.getMessage());
+			retryAndProcessFailedRecords(exp, count, lastReadOffset + 1, eventRecord, consumer, partition,
+					consumerRecord);
+		}
+	}
+
+	
+	private void commitOffsets(TopicPartition partition, long lastOffset, KafkaConsumer<String, String> consumer) {
+		try {
+			long currentTimeMillis = System.currentTimeMillis();
+			if (currentTimeMillis - lastCommitTime > KafkaSourceConfig.MAX_FETCH_TIME
+					&& sourceConfig.getEnableAutoCommit().equalsIgnoreCase(sourceConfig.getEnableAutoCommit())) {
+				consumer.commitSync(Collections.singletonMap(partition, new OffsetAndMetadata(lastOffset + 1)));
+			}
+			lastCommitTime = currentTimeMillis;
+		} catch (CommitFailedException e) {
+			logger.error(KafkaSourceConfig.COMMIT_FAILURE_MSG, e);
+		}
+	}
+
+	
+	private boolean isTopicAvailable(String topic) {
+		Properties config = new Properties();
+		config.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, sourceConfig.getBootStrapServers());
```

Comment:
Why is this put here rather than in the `KafkaSource` along with the other properties?